contracts/base/LoansBase.sol

```diff
@@ -383,6 +396,10 @@ contract LoansBase is LoansInterface, Base {
             EscrowInterface(loans[loanID].escrow).initialize(address(this), loanID);
         }
 
+        if (loans[loanID].userId != bytes32(0)) {
+            userIdHasLoan[loans[loanID].userId] = true;
+        }
```

Comment:
I assume that because the `loans[loanID].userId` value is set it meas that the loan is unsecured? Lets also add the check `!loans[loanID].isLoanSecured()` which is the solidity implementation.